src/server/pps/server/api_server.go

```diff
@@ -828,18 +844,30 @@ func (a *apiServer) getJobDetails(ctx context.Context, jobInfo *pps.JobInfo) err
 
 // ListJob implements the protobuf pps.ListJob RPC
 func (a *apiServer) ListJob(request *pps.ListJobRequest, resp pps.API_ListJobServer) (retErr error) {
-	filter := make(map[string]bool, len(request.GetProjects()))
+	// Prepare filter functions.
+	projectFilter := make(map[string]bool, len(request.GetProjects()))
 	for _, project := range request.GetProjects() {
-		filter[project.GetName()] = true
+		projectFilter[project.GetName()] = true
 	}
-	keep := func(j *pps.JobInfo) error {
-		if len(filter) == 0 || filter[j.Job.Pipeline.Project.GetName()] {
-			return errors.Wrap(resp.Send(j), "could not send filtered job")
+	ctx := resp.Context()
+	jqFilter, err := jqFilterFunc(ctx, request.GetJqFilter())
+	if err != nil {
+		return errors.Wrap(err, "error creating jq filter function")
+	}
+	keep := func(j *pps.JobInfo) (bool, error) {
```

Comment:
I think that this too should be expanded at the call site. It would make the logic clearer too, since both a false or an error lead to immediate returns.